exp/runtime/internal/controllers/extensionconfig_controller.go

```diff
@@ -48,18 +54,103 @@ func (r *Reconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, opt
 	if err != nil {
 		return errors.Wrap(err, "failed setting up with a controller manager")
 	}
+	err = mgr.Add(&warmupRunnable{
+		Client:        r.Client,
+		APIReader:     r.APIReader,
+		RuntimeClient: r.RuntimeClient,
+	})
+	if err != nil {
+		return errors.Wrap(err, "failed adding warmupRunnable to controller manager")
+	}
 	return nil
 }
 
-func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Result, reterr error) {
-	extension := &runtimev1.ExtensionConfig{}
-	err := r.Client.Get(ctx, req.NamespacedName, extension)
+func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	var errs []error
+	// Requeue events when the registry is not ready.
+	if !r.RuntimeClient.IsReady() {
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	extensionConfig := &runtimev1.ExtensionConfig{}
+	err := r.Client.Get(ctx, req.NamespacedName, extensionConfig)
 	if err != nil {
 		if apierrors.IsNotFound(err) {
-			return ctrl.Result{}, nil
+			// ExtensionConfig not found. Remove from registry.
```

Comment:
The issue here is if we somehow miss a delete event we keep the runtime hook in the registry. I'm not sure which of the two edge cases is the more likely/disruptive, but we have more holistic deletion on the task list #6330 so I'm happy to pick one direction or the other for the time being.

Reply:
Let's remove this and go for the holistic solution (a periodic "refresh" of all the extensions)

Reply:
Can we push this to another PR?

Reply:
>  (but at the same time I don't think this should cause problems assuming we handle nicely delete of an extensionConfig not yet registered)

I verified it during my previous review. If we unregister an extension which is not registered it is just a no-op.

Reply:
Though I wonder if this can happen. As we don't use requeue here and in all other controllers this would mean that we could entirely miss create events (which I didn't notice yet)